ballerina/Module.md

```diff
@@ -35,13 +35,36 @@ All trial projects can provision a free trial phone number for testing. Here's h
 ![Search Results](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-twilio/master/ballerina/resources/search-phone-number.png)
 > **Notice:** Many countries require identity documentation for Phone Number compliance. Requests to provision phone numbers with these regulations will be required to select or add the required documentation after clicking Buy in Console. To see which countries and phone number types are affected by these requirements, please see twilio's [Phone Number Regulations](https://www.twilio.com/guidelines/regulatory) site.
 
-### Step 3: Obtain a Twilio account Sid with auth token
+### Step 3: Obtain a Twilio API Key, API Secret with Account SID
```

Comment:
It seems like you have remove Twilio auth-token based authentication section from the setup section. IMO there should be two separate sub-sections in the setup section for auth-token based authentication and API Key-secret based authentication.

Any reason to deviate from that ?

Reply:
Since in each module we are using only the recommended way for the setup guide, instead of the auth token based authentication, I added the API Key based authentication for the setup guide which is the recommended approach by Twilio.

Since the setup guide and the quickstart is there for users to initialise a project with minimal friction, I don't think including multiple authentication methods to the quickstart guide is the best practice. Quickstart and Setup guides are there to be simple and direct, so having multiple authentication mechanisms might be challenging for the users especially the new ones.. WDYT @ayeshLK @NipunaRanasinghe @RDPerera

Reply:
I am not sure this is the correct thing to do. Is there any official documentation for Twilio suggesting that this is the recommended approach ? Unless there is something like that we should not omit it.

Reply:
You can find the recommended approach in the Twilio REST API Best Practices Guide[[1]](https://www.twilio.com/docs/usage/rest-api-best-practices#control-access-to-your-accounts) and in the API Key Docs[[2]](https://www.twilio.com/docs/iam/api-keys)

[1] - [REST API: Best Practices - Control access to your accounts](https://www.twilio.com/docs/usage/rest-api-best-practices#control-access-to-your-accounts)
[2] - [API Keys Overview](https://www.twilio.com/docs/iam/api-keys)

FYI @ayeshLK

Reply:
+1 to mention API key based approach as the convention is to mention about the most secure/recommended authentication mechanism in the setup guide.

Reply:
+1 for going with API Key-based authentication as the main authentication.

However, even though the API-Key-based method is the recommended approach, Account SID and Auth Token are readily available when new users set up a Twilio account(in the Twilio Console). Obtaining them is simple and easy for beginners, so IMO, mentioning auth-token-based authentication is also helpful for beginners.

Seems, libraries from other vendors are also following the hybrid approach;

Python:  https://www.twilio.com/docs/libraries/reference/twilio-python/ (Both auths are documented).
NodeJs: https://www.npmjs.com/package/twilio (Only Auth-Token based auth Documented).
Go: https://github.com/twilio/twilio-go (Only Auth-Token base auth Documented).

Reply:
+1 for mentioning auth-token based authentication as an alternative approach as in Python Twilio Helper Library

Reply:
Added auth token based authentication as an alternative approach in the quickstart by this commit: [0324da8](https://github.com/ballerina-platform/module-ballerinax-twilio/pull/239/commits/0324da821ff6d7eafd31753b683e14b160917069)

Reply:
Don't we need to mention this in the setup guide as well ?

Reply:
We have only focused API key based authentication in this setup guide since it is the recommended/secure approach. However in the quickstart I have mentioned the auth token based authentication as an alternative approach and have added the link to Twilio console where the users can easily retrieve their credentials required for the auth token based authentication. 

I think the setup guide should include one direct way of how to setup a project and get started easily and quickly.. WDYT? @ayeshLK

Reply:
But we can have an alternative approach as a sub section with a note to the setup guide itself. Having links in the Quickstart guide is not enough and that is why we moved into a UI screenshots based approach in the first place.

Reply:
I also think we should mention auth token apporach in the setup guide too, but no need to have a descriptive flow (like we have for the api key authentication). Just mentioning it at the end would be enough as having 2 descriptive flows to setup might confuse users. 
We have followed the same approach in similar scenarios when it comes to other connectors (e.g. [Twitter](https://github.com/ballerina-platform/module-ballerinax-twitter/blob/04c928d11d137bbae04aba171dcdcb20d1021afc/README.md?plain=1#L123))

Reply:
I have updated the Twilio examples with the latest authentication configurations. As discussed, I have also updated three Twilio examples with the auth token based authentication configurations including their setup guides as well.
Please review :) @NipunaRanasinghe @ayeshLK @RDPerera